File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        for i in range(len(capacity)):
            capacity[i] -= rocks[i]
        capacity.sort()
        for i in range(len(capacity)):
            additionalRocks -= capacity[i]
            if additionalRocks == 0:
                return i+1
            if additionalRocks < 0:
                return i
        return len(rocks)

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.php
class Solution {

    /**
     * @param Integer[] $capacity
     * @param Integer[] $rocks
     * @param Integer $additionalRocks
     * @return Integer
     */
    function maximumBags($capacity, $rocks, $additionalRocks) {
        for ($i = 0; $i < count($capacity); $i++) {
            $capacity[$i] -= $rocks[$i];
        }
        sort($capacity);
        for ($i = 0; $i < count($capacity); $i++) {
            $additionalRocks -= $capacity[$i];
            if ($additionalRocks === 0) {
                return $i + 1;
            }
            if ($additionalRocks < 0) {
                return $i;
            }
        }
        return count($rocks);
    }
}